src/matmul.jl

```diff
@@ -59,12 +59,14 @@ end
 function matmul!(dest::GPUArray, a::GPUArray{T, 2}, b::GPUArray{T, 2}) where T
     Asize = size(a)
     Bsize = size(b)
-    TS = UInt32(32)
+    device = GPUArrays.device(a)
+    thr = GPUArrays.threads(device)
+    TS = Int(sqrt(thr))
```

Comment:
needs to be `ceil(Int, sqrt(thr))` for correctness.. Although I suppose, max threads should always be powers of 2